src/toxencryptsave/encryptsave.rs

```diff
@@ -0,0 +1,165 @@
+/*
+    Copyright © 2016 quininer kel <quininer@live.com>
+
+    This file is part of Tox.
+
+    Tox is libre software: you can redistribute it and/or modify
+    it under the terms of the GNU General Public License as published by
+    the Free Software Foundation, either version 3 of the License, or
+    (at your option) any later version.
+
+    Tox is distributed in the hope that it will be useful,
+    but WITHOUT ANY WARRANTY; without even the implied warranty of
+    MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+    GNU General Public License for more details.
+
+    You should have received a copy of the GNU General Public License
+    along with Tox.  If not, see <http://www.gnu.org/licenses/>.
+*/
+
+
+use sodiumoxide::crypto::pwhash::{
+    SALTBYTES, MEMLIMIT_INTERACTIVE, OPSLIMIT_INTERACTIVE,
+    Salt, OpsLimit,
+    gen_salt, derive_key
+};
+use sodiumoxide::crypto::box_::{
+    PRECOMPUTEDKEYBYTES, NONCEBYTES, MACBYTES,
+    Nonce, PrecomputedKey,
+    gen_nonce
+};
+use sodiumoxide::crypto::hash::sha256;
+use ::toxcore::crypto_core;
+
+pub const MAGIC_LENGTH: usize = 8;
```

Comment:
https://en.wikipedia.org/wiki/Magic_number_(programming)
MAGIC_NUMBER is reasonable.

在 2016年05月19日 23:56, zetok 写道:
>
> In src/toxencryptsave/encryptsave.rs
> <https://github.com/zetok/tox/pull/9#discussion_r63905323>:
>
> > +
> > +
> > +use sodiumoxide::crypto::pwhash::{
> > +    SALTBYTES, MEMLIMIT_INTERACTIVE, OPSLIMIT_INTERACTIVE,
> > +    Salt, OpsLimit,
> > +    gen_salt, derive_key
> > +};
> > +use sodiumoxide::crypto::box_::{
> > +    PRECOMPUTEDKEYBYTES, NONCEBYTES, MACBYTES,
> > +    Nonce, PrecomputedKey,
> > +    gen_nonce
> > +};
> > +use sodiumoxide::crypto::hash::sha256;
> > +use ::toxcore::crypto_core;
> > +
> > +pub const MAGIC_LENGTH: usize = 8;
>
> |s/MAGIC_LENGTH/ENCRYPTION_IDENT_LEN/| ?
>
> |ENC / ENCR||instead of|ENCRYPTION` wouldn't be better. right?
>
> —
> You are receiving this because you authored the thread.
> Reply to this email directly or view it on GitHub
> <https://github.com/zetok/tox/pull/9/files/e2a279ff3c6d2be9e0d2e5e167e5302cf9415b47#r63905323>
>